### ISSUE TITLE
Unable to call 'yo.cmd' command in Java after upgrading yeoman to 1.3.3 (Windows)

### ISSUE THREAD

**Issue by RollandLiu** [yeoman/environment#29]
In my project, I need to call the yeoman command in a Java class. It worked fine before(version 1.2.0) but after I upgrade it to 1.3.3, I get an error when running the 'yo.cmd' command. The yeoman command works fine when running it in a command windows directly.

Env: 
    Windows 7 64bit
    Java 1.7.0_51 64bit
    node v0.10.36
    yo 1.3.3

When running the test code: java RunYo, the error message is:

```bash
fs.js:439
  return binding.open(pathModule._makeLong(path), stringToFlags(flags), mode);
                 ^
Error: ENOENT, no such file or directory 'C:\dev\tty'
    at Object.fs.openSync (fs.js:439:18)
    at Object.<anonymous> (C:\Users\myuser\AppData\Roaming\npm\node_modules\yo\node_modules\yeoman-environment\node_modules\inquirer\node_modules\ttys\index.js:9:18)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (C:\Users\myuser\AppData\Roaming\npm\node_modules\yo\node_modules\yeoman-environment\node_modules\inquirer\lib\ui\baseUI.js:8:12)
    at Module._compile (module.js:456:26)
```

Below is the java test code:
---------------------------------------------------------------------------------------------------

```java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.PrintStream;

public class RunYo {

    public static void main(String[] args) {
        String cmd = "yo.cmd";

        try {
            Process process = Runtime.getRuntime().exec(cmd, null, null);
            readProcessOutput(process);
        } catch( IOException e ) {
            e.printStackTrace();
        }
    }

    private static void readProcessOutput(final Process process) {
        read(process.getInputStream(), System.out);
        read(process.getErrorStream(), System.err);
    }

    private static void read(InputStream inputStream, PrintStream out) {
        try {
            BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));

            String line;
            while( (line = reader.readLine()) != null ) {
                out.println(line);
            }

        } catch( IOException e ) {
            e.printStackTrace();
        } finally {

            try {
                inputStream.close();
            } catch( IOException e ) {
                e.printStackTrace();
            }
        }
    }

}
```

-----------------------------------------------------

There is a similar issue in stack overflow reported recently:
http://stackoverflow.com/questions/28077848/windows-7-yeoman-generator-gulp-angular-not-finding-creating-directory-erro

**Comment by SBoudrias** [comment]
This have been fixed a while ago. Cache clean, update, uninstall/reinstall whatev.

**Comment by RollandLiu** [comment]
Hi Simon, I tired the cache clean and re-installing the 1.3.3 but still got the same error. I also upgraded the npm to 2.4.1 and install yo@1.4.5 but it didn't work. The yo@1.2.0 works. 

Did you try calling yo from Java, and could you please take a look at it again?

Thanks,
Rolland

**Comment by kpiascik** [comment]
@SBoudrias , can you please reference which version has the fix and if there's a github issue that is related to the fix?

**Comment by SBoudrias** [comment]
It's a bug with Inquirer.js when runned in non-tty terminals. If you use inquirer 0.8.0 you won't have the issue.

**Comment by RollandLiu** [comment]
I tested the inquirer 0.8.0 and it worked fine, but the default dependency of yo is 0.8.1. We need to fix this. Thanks.

**Comment by SBoudrias** [comment]
@RollandLiu It is not: https://github.com/yeoman/yo/blob/master/package.json#L52 - and there's no inquirer with higher version than 0.8.0 published to npm. Maybe you use a local server or a mirror?

**Comment by RollandLiu** [comment]
Yes, we do have a internal server and it's using the non-published inquirer 0.8.1. Thank you so much for your help.
